File: 1666-make-the-string-great/make-the-string-great.py
class Solution(object):
    def makeGood(self, s):
        """
        :type s: str
        :rtype: str
        """
        stack = []
        
        for char in s:
            if stack and abs(ord(char) - ord(stack[-1])) == 32:
                stack.pop()  # Remove the last character from the stack
            else:
                stack.append(char)  # Add the current character to the stack
        
        return ''.join(stack)

File: 1666-make-the-string-great/make-the-string-great.java
class Solution {
    public String makeGood(String s) {
        StringBuilder sb = new StringBuilder();
        
        for (char c : s.toCharArray()) {
            if (sb.length() > 0 && Math.abs(sb.charAt(sb.length() - 1) - c) == 32) {
                sb.deleteCharAt(sb.length() - 1); // Remove the last character from the StringBuilder
            } else {
                sb.append(c); // Append the current character to the StringBuilder
            }
        }
        
        return sb.toString();
    }
}